go/libraries/doltcore/sqle/binlogreplication/binlog_replication_reconnect_test.go

```diff
@@ -40,15 +41,20 @@ func TestBinlogReplicationAutoReconnect(t *testing.T) {
 	configureFastConnectionRetry(t)
 	startReplication(t, proxyPort)
 
+	// Get the replica started up and ensure it's in sync with the primary before turning on the limit_data toxic
 	testInitialReplicaStatus(t)
-
 	primaryDatabase.MustExec("create table reconnect_test(pk int primary key, c1 varchar(255));")
+	waitForReplicaToCatchUp(t)
+	turnOnLimitDataToxic(t)
+
 	for i := 0; i < 1000; i++ {
 		value := "foobarbazbashfoobarbazbashfoobarbazbashfoobarbazbashfoobarbazbash"
 		primaryDatabase.MustExec(fmt.Sprintf("insert into reconnect_test values (%v, %q)", i, value))
 	}
 	// Remove the limit_data toxic so that a connection can be reestablished
-	mysqlProxy.RemoveToxic("limit_data")
+	err := mysqlProxy.RemoveToxic("limit_data")
+	require.NoError(t, err)
+	fmt.Printf("Toxiproxy proxy limit_data toxic removed")
```

Comment:
Nit: `t.Logf(...)` will ensure this prints if `-test.v` is supplied or anytime the test fails, which can be nice.

Reply:
ooh... thanks for pointing that out – I hadn't come across that yet. I went ahead and switched this file over to `t.Logf` and will prefer that func in the future, too.